workspace_test.go

```diff
@@ -1066,6 +1100,130 @@ func TestWorkspaces_UpdateRemoteStateConsumers(t *testing.T) {
 	})
 }
 
+func TestWorkspaces_AddTags(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	orgTest, orgTestCleanup := createOrganization(t, client)
+	defer orgTestCleanup()
+
+	wTest, wTestCleanup := createWorkspace(t, client, orgTest)
+	defer wTestCleanup()
+
+	options := WorkspaceAddTagsOptions{
+		Tags: []*Tag{
+			{
+				Name: "tag1",
+			},
+			{
+				Name: "tag2",
+			},
+			{
+				Name: "tag3",
+			},
+		},
+	}
+
+	t.Run("successfully adds tags", func(t *testing.T) {
+		err := client.Workspaces.AddTags(ctx, wTest.ID, options)
+		require.NoError(t, err)
+
+		w, err := client.Workspaces.Read(ctx, orgTest.Name, wTest.Name)
+		require.NoError(t, err)
+		assert.Equal(t, 3, len(w.TagNames))
+		assert.Equal(t, w.TagNames, []string{"tag1", "tag2", "tag3"})
+
+		err = client.Workspaces.AddTags(ctx, wTest.ID, WorkspaceAddTagsOptions{
+			Tags: []*Tag{
+				{
+					Name: "tag4",
+				},
+			},
+		})
+		require.NoError(t, err)
+
+		w, err = client.Workspaces.Read(ctx, orgTest.Name, wTest.Name)
+		require.NoError(t, err)
+		assert.Equal(t, 4, len(w.TagNames))
+		assert.Equal(t, w.TagNames, []string{"tag1", "tag2", "tag3", "tag4"})
+	})
+
+	t.Run("with invalid options", func(t *testing.T) {
+		err := client.Workspaces.AddTags(ctx, wTest.ID, WorkspaceAddTagsOptions{
+			Tags: []*Tag{},
+		})
+		require.Error(t, err)
+		assert.EqualError(t, err, ErrMissingTagIdentifier.Error())
+	})
+
+	t.Run("without a valid workspace ID", func(t *testing.T) {
+		err := client.Workspaces.AddTags(ctx, badIdentifier, WorkspaceAddTagsOptions{})
+		require.Error(t, err)
+		assert.EqualError(t, err, ErrInvalidWorkspaceID.Error())
+	})
+}
```

Comment:
Can we add one more `t.Run` test here for handling duplicates?

```
	t.Run("successfully handles duplicate tags to be added", func(t *testing.T) {
		err := client.Workspaces.AddTags(ctx, wTest.ID, options)
		require.NoError(t, err)

		w, err := client.Workspaces.Read(ctx, orgTest.Name, wTest.Name)
		require.NoError(t, err)
		assert.Equal(t, 3, len(w.TagNames))
		assert.Equal(t, w.TagNames, []string{"tag1", "tag2", "tag3"})

		err = client.Workspaces.AddTags(ctx, wTest.ID, WorkspaceAddTagsOptions{
			Tags: []*Tag{
				{
					Name: "tag3",
				},
			},
		})
		require.NoError(t, err)

		w, err = client.Workspaces.Read(ctx, orgTest.Name, wTest.Name)
		require.NoError(t, err)
		assert.Equal(t, 3, len(w.TagNames))
		assert.Equal(t, w.TagNames, []string{"tag1", "tag2", "tag3"})
	}
```

Reply:
wouldn't that be more a function of the API server rather than this client? or is there some mechanism in go-tfe we would test with this?

Reply:
> wouldn't that be more a function of the API server rather than this client
Fair point 👍